### PR TITLE
Update/add at guard to themes showcase

### PR DESCRIPTION
**Pull request by budzanowski**
### Info
We want to be able to make Jetpack upload live and at the same time block AT from happening until AT config flag is switched on for production. Right now our internal flag `manage/themes/upload` stops us from doing that. The solution is to use `automated-transfer` flag to block upload page and hide upload button for all scenarios except Jetpack. This way when `manage/themes/upload` will be switched on in production only Jetpack will have upload enabled. 

When upload is not available `design/upload` looks like this:
image

@folletto ^ ?

### Testing
When only `manage/themes/upload`  is enabled Upload button and upload page work only for Jetpack
When also `automated-transfer` flag is enabled Upload button and page work for all sites.

### CODE REVIEW

**Comment by folletto** [comment]
Copy:

* Title: Upload not available for this site
* Subtitle: Please select a different site
* Button: Back to themes

File: client/my-sites/themes/theme-showcase.jsx [hunk 1]
```diff
@@ -142,13 +153,14 @@ const ThemeShowcase = React.createClass( {
...
 				</StickyPanel>
 				{ config.isEnabled( 'manage/themes/upload' ) && siteSlug && ! isMultisite &&
-					<Button className="themes__upload-button" compact icon
-						onClick={ this.onUploadClick }
-						href={ `/design/upload/${ this.props.siteSlug }` }
-					>
-						<Gridicon icon="cloud-upload" />
-						{ translate( 'Upload Theme' ) }
-					</Button>
+					( isJetpack || config.isEnabled( 'automated-transfer' ) ) &&
```
Comment (seear) [inline client/my-sites/themes/theme-showcase.jsx]: Maybe all this logic should be moved into a separate function?

File: client/my-sites/themes/theme-showcase.jsx [hunk 2]
```diff
@@ -118,17 +118,21 @@ const ThemeShowcase = React.createClass( {
 		trackClick( 'upload theme' );
 	},
 
+	showUploadButton() {
+		const { siteSlug, isMultisite, isJetpack } = this.props;
+
+		return config.isEnabled( 'manage/themes/upload' ) && siteSlug && ! isMultisite &&
+			( isJetpack || config.isEnabled( 'automated-transfer' ) );
```
Comment (seear) [inline client/my-sites/themes/theme-showcase.jsx]: It's taken me a while to get used to formatting long boolean clauses like this in an extra set of `()` like this:

```js
return(
    config.isEnabled( 'manage/themes/upload' ) &&
    siteSlug &&
    ! isMultisite &&
    ( isJetpack || config.isEnabled( 'automated-transfer' ) )
);
```

Comment (seear) [inline client/my-sites/themes/theme-showcase.jsx]: Alternatively, you can do it as a `const` inside `render()`, meaning you get a nice `{ showUploadButton && <Button...` without needing `this.showUp...`

```js
const showUploadButton = (
    config.isEnabled( 'manage/themes/upload' ) &&
    siteSlug &&
    ! isMultisite &&
    ( isJetpack || config.isEnabled( 'automated-transfer' ) )
);
```

File: client/my-sites/themes/theme-upload/index.jsx [hunk 3]
```diff
@@ -279,6 +280,19 @@ class Upload extends React.Component {
 			/>
 			);
 	}
+
+	renderNotAvailable() {
+		return (
+			<EmptyContent
+				title={ this.props.translate( 'Upload not available for this site' ) }
+				line={ this.props.translate( 'Please select a different site' ) }
+				action={ this.props.translate( 'Backt to themes' ) }
```
Comment (deBhal) [inline client/my-sites/themes/theme-upload/index.jsx]: A little bit late, but there's a typo here

**Pull request merged**